tracing/src/span.rs

```diff
@@ -379,6 +380,7 @@ pub(crate) struct Inner {
 #[must_use = "once a span has been entered, it should be exited"]
 pub struct Entered<'a> {
     span: &'a Span,
+    _not_send: PhantomData<*mut ()>,
```

Comment:
```suggestion
    _not_send_but_sync: PhantomData<std::sync::MutexGuard<'static, ()>>,
```
It's a bit weird, but -1 unsafe.

Reply:
oh that's clever! thanks!

unfortunately we can't do *exactly* what you've suggested, because this needs to compile with the `std` feature disabled, but I can do something similar.

Reply:
This isn't actually unsafe? We are not dereferencing this type, thus doesn't require unsafe.

Reply:
@LucioFranco the unsafe was necessary for manually implementing `Sync`.

Reply:
...and, it turns out this approach doesn't actually work in this specific case. Because `tracing` has conditional support for `no_std`, we can't use the `std::sync` version of `MutexGuard`. I thought we could use the `MutexGuard` from the spinlock implementation that's used as a fallback on no-std platforms, but I'd forgotten that this is defined in `tracing-core`, not `tracing`, and isn't publicly visible. I'll probably go back to the previous approach of just accepting the additional `unsafe` keyword necessary to manually impl `Sync`.